src/App.tsx

```diff
@@ -1,29 +1,57 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/indent */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const TOTAL_ITEMS = 42;
+const items = getNumbers(1, TOTAL_ITEMS).map(n => `Item ${n}`);
+const itemsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(itemsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const itemsFrom = perPage * currentPage - perPage + 1;
+  const itemsTo = Math.min(perPage * currentPage, TOTAL_ITEMS);
+
+  const visibleItems = items.filter(
+    (_item, index) => index >= itemsFrom - 1 && index <= itemsTo - 1,
+  );
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
```

Comment:
When changing the number of items per page, it might be beneficial to adjust the current page to ensure the user remains in a relevant position in the list. For instance, if the user is on page 4 with 5 items per page and switches to 10 items per page, they might expect to still see some of the items that were on page 4. Consider calculating the new page based on the first visible item's index.